app/descriptor/views.py

```diff
@@ -243,48 +241,35 @@ def remove_option_value(desc_id, option_index):
 
     # If no resources are affected, just remove the option value.
     if len(option_assocs) == 0:
-        # Index starting from 1 to skip 'Remove this descriptor'
-        remove_value_from_db(descriptor, choice_names[1:], old_value)
+        remove_value_from_db(descriptor, choice_names, old_value)
         return redirect(url_for('descriptor.descriptor_info', desc_id=desc_id))
 
     # Create the select field for each resource.
-    for oa in option_assocs:
-        setattr(FixAllResourceOptionValueForm, oa.resource.name,
-                SelectField('', coerce=int, choices=choices))
     form = FixAllResourceOptionValueForm()
 
     # Delete the dynamic fields after the form is instantiated
-    for oa in option_assocs:
-        delattr(FixAllResourceOptionValueForm, oa.resource.name)
-
     if form.validate_on_submit():
         for oa in option_assocs:
-            choice = form[oa.resource.name].data
             # Case for 'Remove this descriptor'
-            if choice == -1:
-                db.session.delete(oa)
-            else:
-                oa.option = choice
-                db.session.add(oa)
-
-        # Index starting from 1 to skip 'Remove this descriptor'
-        if remove_value_from_db(descriptor, choice_names[1:], old_value):
+            db.session.delete(oa)
+            print oa
+
+        if remove_value_from_db(descriptor, choice_names, old_value):
             return redirect(url_for('descriptor.descriptor_info',
                                     desc_id=desc_id))
     return render_template('descriptor/confirm_resources.html',
                            option_assocs=option_assocs, desc_id=desc_id,
                            desc=descriptor, option_index=option_index,
                            form=form)
 
-
 def generate_option_choices(descriptor, removed_index):
-    """Helper function to generate the option values for a SelectField."""
-    choice_names = (['Remove this descriptor'] +
-                    descriptor.values[:removed_index] +
+    """Helper function to generate the option values for a SelectMultipleField."""
```

Comment:
we can reword this comment since i don't think we're using these values for any type of SelectField?